### PR TITLE
Make DefaultTask.getName() final

### LINKED ISSUE #19538: Make  Task.getName() final
Right now it's possible to override `Task.getName` method. This should not be allowed, since in case the user misconfigures this, the task cannot be found. There is actually no good use case to allow this. 

---
cc: @gradle/execution

### PR DESCRIPTION
**Pull request by asodja**
Fixes #19538

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
It fails because we mock `DefaultTask` in `DefaultTaskContainerTest`. Auch.

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/core-api/src/main/java/org/gradle/api/Task.java [hunk 1]
```diff
@@ -181,6 +181,8 @@
     /**
      * <p>Returns the name of this task. The name uniquely identifies the task within its {@link Project}.</p>
      *
+     * <p>This method should never be implemented by a custom Task.</p>
```
Comment (wolfs) [inline subprojects/core-api/src/main/java/org/gradle/api/Task.java:184]: This is true for all methods on this interface. Maybe we should move it to the class Javadoc?

File: subprojects/core/src/main/java/org/gradle/api/DefaultTask.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public Project getProject() {
     }
 
     @Override
-    public String getName() {
+    public final String getName() {
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/DefaultTask.java:63]: It seems somewhat unmotivated to sprinkle `final` on the methods in `DefaultTask`. I wonder if there is a better way to ensure nobody overrides these methods.

File: subprojects/core-api/src/main/java/org/gradle/api/Task.java (hunk 1 above)
Comment (asodja) [reply subprojects/core-api/src/main/java/org/gradle/api/Task.java:184]: Makes sense, I will move to the class Javadoc

File: subprojects/core/src/main/java/org/gradle/api/DefaultTask.java (hunk 2 above)
Comment (asodja) [reply subprojects/core/src/main/java/org/gradle/api/DefaultTask.java:63]: I will check if we can do this when we decorate a class or somewhere there

**Comment by wolfs** [comment]
Let's close this for now until we come back to it.

**Pull request closed without merging**